### PR TITLE
Checkin function group in abapGit format

### PR DESCRIPTION
**Pull request by palubaj**

### CODE REVIEW

File: sap/cli/checkin.py [hunk 1]
```diff
@@ -415,13 +415,96 @@ def _check_fugr_source_files(repo_obj, functions, includes):
...
     """Write source file for ADT object"""
 
     adt_object_file_path = path_prefix + f'.{adt_object.name.lower()}' + '.abap'
     with open(adt_object_file_path, 'r', encoding='utf-8') as source:
-        with adt_object.open_editor(corrnr=corrnr) as editor:
-            editor.write(source.read())
+        _write_source_file(source.read(), adt_object, corrnr)
+
+
+def _get_parameters_block(source_lines):
```
Comment (jfilak) [inline sap/cli/checkin.py:431]: Please add some text graphic explaining the logic.

File: sap/cli/checkin.py [hunk 2]
```diff
@@ -415,13 +415,96 @@ def _check_fugr_source_files(repo_obj, functions, includes):
...
+    for i, line in enumerate(source_lines[start_block + 1:]):
+        if line.startswith('*"--'):
+            end_block = i + start_block + 1
+            break
+
+    return start_block, end_block
+
+
+def _parse_function_parameters(parameters_block):
+    """Parse parameters of function module"""
```
Comment (jfilak) [inline sap/cli/checkin.py:450]: Please add a text graphic explaining the logic.

File: sap/cli/checkin.py [hunk 3]
```diff
@@ -415,13 +415,96 @@ def _check_fugr_source_files(repo_obj, functions, includes):
...
+    for i, table_param in enumerate(parameters['TABLES']):
+        var_name, abap_typing, *abap_type = table_param.split(' ')
+        if abap_typing == 'STRUCTURE':
+            parameters['TABLES'][i] = f'{var_name} TYPE {" ".join(abap_type)}'
+
+    return parameters
+
+
+def _format_function(source_code):
+    """Format parameters of function"""
```
Comment (jfilak) [inline sap/cli/checkin.py:476]: Please add a text graphic explaining the logic.

**Comment by jfilak** [comment]
Please squash the commits and add sample input and output to the commit message.

**Comment by palubaj** [comment]
Updated and ready.

File: sap/cli/checkin.py [hunk 4]
```diff
@@ -415,13 +415,154 @@ def _check_fugr_source_files(repo_obj, functions, includes):
...
+    parameters = {
+        'IMPORTING': [],
+        'EXPORTING': [],
+        'CHANGING': [],
+        'TABLES': [],
+        'EXCEPTIONS': []
+    }
+    current_param = None
+    for line in parameters_block:
+        if any(param in line for param in parameters):
```
Comment (jfilak) [inline sap/cli/checkin.py:493]: what if I have a parameter named "READ_TABLES"?

Comment (jfilak) [reply sap/cli/checkin.py:493]: I would use a regex or simply always do `param = line.lstrip('*" ').rstrip()` and use the equality operator for checking if the current line is a "param type" label.

File: test/unit/fixtures_cli_checkin.py [hunk 5]
```diff
@@ -144,3 +144,27 @@
...
+
+FUNCTION_MODULE_BODY_ABAPGIT = '''FUNCTION ztest_function_module.
+*"----------------------------------------------------------------------
+*"*"Local Interface:
+*"  IMPORTING
+*"     VALUE(ABAP_PACKAGE) TYPE  DEVCLASS OPTIONAL
+*"  EXPORTING
+*"     VALUE(RETURN) TYPE  BAPIRET2
+*"  TABLES
+*"      RETURN_TAB STRUCTURE  BAPIRET2 OPTIONAL
```
Comment (jfilak) [inline test/unit/fixtures_cli_checkin.py:156]: This test does not check all types. CHANGING and EXCEPTIONS are missing.

File: test/unit/test_sap_cli_checkin.py [hunk 6]
```diff
@@ -1055,6 +1056,14 @@ def test_checkin_fugr_already_created(self, fake_mod_log):
                                                          call('Function include already exists'),
                                                          call('Function module already exists')])
 
+    def test_checkin_fugr_abapgit_format(self):
```
Comment (jfilak) [inline test/unit/test_sap_cli_checkin.py:1059]: Please add a test for function without any parameters.

**Comment by jfilak** [comment]
Good job! Thank you.

**Pull request merged**